File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int n) {
        int firstmax=Integer.MIN_VALUE;
        int secondmax=Integer.MIN_VALUE;
        int num=n;
        while(num>0){
            int val=num%10;
            if(val>firstmax)firstmax=val;
            num/=10;
        }
        num=n;
        int count=0;
        while(num>0){
            int val=num%10;
            if(val>secondmax && val!=firstmax){
                secondmax=val;
            }
            if(val==firstmax)count++;
            num/=10;
        }
        if(count>1)secondmax=firstmax;
        return firstmax*secondmax;
    }
}

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        first=-1
        second=-1
        num=n
        while (num>0):
            if num%10>first:
                first=num%10
            num/=10
            num=int(num)
        num=n
        c=0
        while num>0:
            val=num%10
            if val>second:
                if val==first:
                    c+=1
                else :
                    second=val
            num/=10
            num=int(num)
        if c>1:
            return first*first
        return first*second